tests/testsuite/a_basic/tests_02_machines.py

```diff
@@ -154,27 +155,92 @@ def test004_pause_resume_get_kvm(self):
         *Test case for testing pausing resuming VMs*
 
         **Test Scenario:**
-        #. Create virtual machine (VM), should succeed
-        #. Pause the VM and check state from get method, should be paused
-        #. Resume the VM and check state from get method, should be resumed
+        #. Create virtual machine (VM1), should succeed
+        #. Pause VM1 and check state from get method, should be paused
+        #. Make sure you can't reach VM1.
+        #. Resume VM1 and check state from get method, should be resumed
+        #. Make sure you can reach the VM1.
         #. Destroy VM1, should succeed
         """
-
         self.lg('{} STARTED'.format(self._testID))
         vm_name = self.rand_str()
         self.lg('- Create virtual machine {} , should succeed'.format(vm_name))
-        vm_uuid = self.create_vm(name=vm_name, flist=self.ubuntu_flist)
-        time.sleep(3)
-
-        self.lg('Pause the VM and check state from get method ,should be paused')
+        pub_key = self.create_ssh_key()
+        nics = [{'type': 'default'}]
+        pub_port = randint(4000, 5000)
+        config = {'/root/.ssh/authorized_keys': pub_key}
+        vm_uuid = self.create_vm(name=vm_name, flist=self.ubuntu_flist,
+                                 config=config, nics=nics, port={pub_port: 22})
+        self.lg('Make sure VM1 is reachable')
+        time.sleep(20)
```

Comment:
i prefer making loop with timeout period 
and within the loop try to access the vm